interview-questions.md

```diff
@@ -15,54 +15,54 @@ INSTRUCTOR EXAMPLE: What is a conditional statement?
 
-  Researched answer:
+  Researched answer: GIT is installed locally to manage source code and its history. Github uses GIT and is a cloud based tool that allows developers to upload and download code from a central location.
 
 
 
 2. Which JavaScript operators will return a Boolean value?
 
-  Your answer:
+  Your answer: AND (&&) and OR (||) will return boolean values. An AND will check to see if both statements are true, while an OR will return a value of true if at least one of the statements is true.
 
-  Researched answer:
+  Researched answer: Another operator that will return a Boolean value is NOT(!). It takes a true statement and makes it false, and vice versa. For example, if you declare a variable (let example) and assign it the value of true, when you call on that variable and use (!example), it would return as false. It can also be used with equality operators. (!==)
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: An index is the location at which a value is found in a string or an array. With zero based index, you start with the first value at index 0, and the next value would be at 1 and so on.
 
-  Researched answer:
+  Researched answer: Index is where you can find a value within a set of data. Value is the data itself that is contained in a certain position (index) in the collection. 
 
 
 
 4. What is iteration?
 
-  Your answer:
+  Your answer: Iteration is the repetition of a block of code. A good example would be loops, which you can code and ask the computer to go through an array and return a value after each iteration. 
 
-  Researched answer:
+  Researched answer: It is the process of executing an action until a condition is met. It is useful for developers in order to keep them from repeating themselves, also known as DRY (dont repeat yourself) principle.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: In pair programming you have the roles of driver and navigator. Both roles are responsible for coming up with a plan. The driver is the one who will be typing and writing code. The role of the navigator is to guide the driver and act as a second pair of eyes for anything that the driver might overlook, such as errors in the code, mispellings, etc. 
 
-  Researched answer:
+  Researched answer: Pair programming allows you to have two developers working together to produce more efficient code. While the driver has control of the keyboard, he is following the navigators guidance. This in turn keeps the navigator engaged instead of the possibility of just taking a backseat in the process and just watching the driver code. One of the most important things in pair programming is constant communication. It is key to talk out your thought process and switch roles often.
```

Comment:
Really great answers in this section!  Very thorough and well thought out on all of these, excellent work!!